cvat/apps/engine/cache.py

```diff
@@ -412,7 +413,11 @@ def get_or_set_task_chunk(
             self._make_chunk_key(db_task, chunk_number, quality=quality),
             set_callback,
         )
-        db_task.refresh_from_db(fields=["segment_set"])
+
+        if is_field_cached(db_task, "segment_set"):
+            # Refresh segments to report actual dates if they were fetched previously
+            # Doing so without a check leads to an error if the related object is not prefetched
```

Comment:
FWIW, I think that's a bug, so I filed a Django bug report: https://code.djangoproject.com/ticket/36372.